hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALSplitter.java

```diff
@@ -1369,13 +1369,17 @@ public LogRecoveredEditsOutputSink(PipelineController controller, EntryBuffers e
      */
     @Override
     public List<Path> finishWritingAndClose() throws IOException {
-      boolean isSuccessful = false;
-      List<Path> result = null;
+      boolean isSuccessful;
+      List<Path> result;
+      List<IOException> thrown;
       try {
         isSuccessful = finishWriting(false);
       } finally {
-        result = close();
```

Comment:
Before this change, in case of exception we are closing the files first and then closing the writers. This will create corrupt wals since the file is closed while the writer is still writing to these files. We might miss some data and close will append the trailer to them.